### DISCUSSION TITLE
[Question] How to break text into words?

### DISCUSSION THREAD

**Discussion by saona-raimundo** [unicode-org/icu4x#2794] [Q&A] [answered]
First of all, thanks for the crate!!

I am looking into coding something like this [ICU demo](https://icu4c-demos.unicode.org/icu-bin/icusegments#3/en) for computing ICU segments using icu4x in Rust, but I could not find how to use the [icu::segmenter::WordBreakSegmenter](https://docs.rs/icu/latest/icu/segmenter/struct.WordBreakSegmenter.html) in a locale-aware fashion. 

Using [unicode_segmentation::UnicodeSegmentation](https://unicode-rs.github.io/unicode-segmentation/unicode_segmentation/trait.UnicodeSegmentation.html#tymethod.unicode_words) is pretty straight forward, but this only implements the [Default Word Boundary Specification](http://www.unicode.org/reports/tr29/#Default_Word_Boundaries). I am looking into doing a similar job being locale-aware.

I know that [icu::segmenter::WordBreakSegmenter](https://docs.rs/icu/latest/icu/segmenter/struct.WordBreakSegmenter.html) is behind the `experimental` flag, but I was wondering if it is possible to do it today(?) Its constructor does not seem to be locale-aware (but maybe I am wrong)

As a benchmark, I would like to break this text in [Primitive Irish](https://en.wikipedia.org/wiki/Primitive_Irish) (locale `pgl`, using the [Ogham alphabet](https://ogham.co/ogham-alphabet/))
```
ᚁᚔ ᚇᚔᚂᚔᚄ ᚇᚒᚔᚈ ᚃᚓᚔᚅ
```
into words `ᚁᚔ`, `ᚇᚔᚂᚔᚄ`, `ᚇᚒᚔᚈ`, and `ᚃᚓᚔᚅ`.

Thanks beforehand!

**Comment by sffc** [comment]
Like ICU4C, ICU4X has dictionary-based segmentation for Chinese, Japanese, Korean, Thai, Burmese, Lao, and Khmer. It also has machine-learning-based segmentation for Thai, Burmese, Lao, and Khmer.

@makotokato @aethanyc Do we support `pgl-Ogam` segmentation?

**Comment by aethanyc** [comment, marked as answer]
> @makotokato @aethanyc Do we support pgl-Ogam segmentation?

No, we don't. We currently only have the same set of dictionaries as ICU4C, so for `pgl-Ogam` segmentation, we'll use [Default Word Boundary Specification](http://www.unicode.org/reports/tr29/#Default_Word_Boundaries) in UAX29.
